### PR TITLE
Allow rename database export file in settings

### LINKED ISSUE #5780: Allow renaming export file
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Could you please add an option to rename the file while exporting? It's frustrating that we can't name it on our own. Every single time, it creates file with a different name.

If a new export file has the same name of any previous file in that folder, a popup should appear like "Np.xml already exists." Three options can be added as reply, "Replace file", "Save as Np1" and "Cancel exporting"

#### How will you/everyone benefit from this feature?

It becomes really difficult as there is no way to replace the previous file. Plus, keeping more files also turns things confusing. Allowing renaming will also let us avoid mixing up files with similar names.

### PR DESCRIPTION
**Pull request by ATofighi**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

In export database on settings. I show a new dialog and get file name from user. then I save export file with entered name.

#### Fixes the following issue(s)

- Fixes #5780 

#### Relies on the following changes

- #6319 

#### APK testing 

On the website the APK can be found by going to the "Checks" tab below the title and then on "artifacts" on the right.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SameenAhnaf** [comment]
Thank you so much for all the efforts. Works great on my device.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java [hunk 1]
```diff
@@ -195,6 +197,33 @@ public void onActivityResult(final int requestCode, final int resultCode,
...
+        fileNameET.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_URI);
+        fileNameET.setHint(R.string.export_data_file_name_hint);
+
+        final AlertDialog dialog = new AlertDialog.Builder(c)
+                .setTitle(R.string.export_data_file_name)
+                .setIcon(R.drawable.ic_import_export)
+                .setNegativeButton(R.string.cancel, null)
+                .setPositiveButton(R.string.export_button, (dialog1, which) -> {
+                    final String fileName = fileNameET.getText().toString();
+                    if (FilenameUtils.isValidFileName(fileName, "zip")) {
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:215]: We could also append ".zip" when the file name does not have that extension.

Comment (ATofighi) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:215]: So, should I don't ask extension of file in file name or if user enter file name without any dots, I append ".zip" at the end of it?

What should I do if user enter "foo.rar" ? make it "foo.zip"? or "foo.rar.zip"?

**Comment by Stypox** [comment]
Note: don't merge this before #5415

**Comment by Stypox** [comment]
There is no need for this anymore, as SAF supports renaming while creating files

**Pull request closed without merging**